components/exchangeMenu/ExchangeButton.tsx

```diff
@@ -0,0 +1,54 @@
+import { Icon } from '@makerdao/dai-ui-icons'
+import { useOnClickOutside } from 'helpers/useOnClickOutside'
+import React, { useRef, useState } from 'react'
+import { Box, Button, Flex, SxStyleProp } from 'theme-ui'
+import { provider } from 'web3-core'
+
+import { UniswapWidget } from './UniswapWidget'
+
+export function ExchangeButton({
+  web3Provider,
+  sx,
+}: {
+  web3Provider?: provider
+  sx?: SxStyleProp
+}) {
+  const componentRef = useRef(null)
+  const [isOpen, setIsOpen] = useState(false)
+
+  useOnClickOutside(componentRef, () => setIsOpen(false))
+
+  return (
+    <Flex ref={componentRef} sx={{ position: 'relative', mr: 2, pr: 1, ...sx }}>
```

Comment:
I love this hook you've made.  Could be like this and return the ref from the hook?

```suggestion
  const componentRef = useOnClickOutside(() => setIsOpen(false))
  const [isOpen, setIsOpen] = useState(false)

  return (
    <Flex ref={componentRef} sx={{ position: 'relative', mr: 2, pr: 1, ...sx }}>
```

Reply:
Woah, nice, I'll try it.

Reply:
Done.